### PR TITLE
[SCB-1562]add oauth2 authorization module

### PR DESCRIPTION
**Pull request by SmartsYoung**
Modify the PR #84 content and rebuild code.

### CODE REVIEW

**Comment by SmartsYoung** [comment]
@tianxiaoliang Already refactored code.

File: proxy/handler/oauth2/oauth2_handler.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+const AuthName = "oauth2"
+
+//Handler is is a oauth2 pre process raw data in handler
+type Handler struct {
+}
+
+// Handle is provider
+func (oa *Handler) Handle(chain *handler.Chain, inv *invocation.Invocation, cb invocation.ResponseCallBack) {
+	if req, ok := inv.Args.(*http.Request); ok {
+		grandType := req.FormValue("grand_type")
```
Comment (ChinX) [inline proxy/handler/oauth2/oauth2_handler.go]: Misspelled, it should be "grant_type"

Comment (SmartsYoung) [reply proxy/handler/oauth2/oauth2_handler.go]: Thanks, it's already fix it.

File: docs/oauth2/oauth2.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+Writing business code
+
+When you use authorization code model, you need to implement the follow parameters. Otherwise you need to implement the interface of config in /oauth2manage.
+
+For example, implement the authorization code model **In config.go**
+
+```go
+
+	auth.Use(&auth.Config{
+		UseConfig: func() *oauth2.Config {
```
Comment (tianxiaoliang) [inline docs/oauth2/oauth2.md]: 调用起来复杂度有点搞了，如果只是设置配置的话，这样不是更简单么

authSetOauth2Config( *oauth2.Config {

**Pull request closed without merging**